src/e2e-test/java/io/cdap/plugin/servicenowsink/actions/ServiceNowSinkPropertiesPageActions.java

```diff
@@ -53,11 +62,93 @@ public static void verifyIfRecordExistsInServiecNowReceivingSlipLineTable(String
       "", "", "");
 
     ServiceNowTableAPIClientImpl tableAPIClient = new ServiceNowTableAPIClientImpl(config);
-    responseStatusCodeIsSuccess = tableAPIClient.verifyIfRecordInServiceNowTableExists(
-      TablesInTableMode.RECEIVING_SLIP_LINE.value, query);
+    responseFromServiceNowTable = tableAPIClient.getRecordFromServiceNowTable(tableName, query);
+  }
+
+  public static void verifyIfRecordCreatedInServiceNowIsCorrect(String query, String tableName)
+    throws IOException, InterruptedException, OAuthProblemException, OAuthSystemException {
 
-    logger.info("Verifying that the Response Status Code is true: " + responseStatusCodeIsSuccess);
-    Assert.assertTrue(responseStatusCodeIsSuccess);
+    getRecordFromServiceNowTable(query, tableName);
+    TableResult bigQueryTableData = getBigQueryTableData(TestSetupHooks.bqSourceDataset, TestSetupHooks.bqSourceTable);
+    String bigQueryJsonResponse = bigQueryTableData.getValues().iterator().next().get(0).getValue().toString();
+    Gson gson = new Gson();
+    JsonObject jsonObject = gson.fromJson(bigQueryJsonResponse, JsonObject.class);
+    Map<String, Object> bigQueryResponseInMap = gson.fromJson(jsonObject.toString(), Map.class);
+
+    logger.info("Response of BigQuery : " + bigQueryResponseInMap);
+    logger.info("Response of ServiceNow : " + responseFromServiceNowTable);
+    Assert.assertTrue(compareValueOfBothResponses(responseFromServiceNowTable, bigQueryResponseInMap));
+  }
+
+  public static void verifyIfRecordUpdatedInServiceNowIsCorrect(String query, String tableName)
+    throws IOException, InterruptedException, OAuthProblemException, OAuthSystemException {
+
+    getRecordFromServiceNowTable(query, tableName);
+    TableResult bigQueryTableData = getBigQueryTableData(TestSetupHooks.bqSourceDataset, TestSetupHooks.bqSourceTable);
+    String bigQueryJsonResponse = bigQueryTableData.getValues().iterator().next().get(0).getValue().toString();
```

Comment:
this code can throw NullPointerException if table contains no data, you can add a null check to be on the safer side.

Reply:
Added a check.